web/app/scripts/utils/date-utils.js

```diff
@@ -0,0 +1,87 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DateUtils(LanguageState) {
+        var languageMap = {
+            'ar-sa': {
+                language: 'ar',
+                calendar: 'ummalqura',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'd MM, Y',
+                    'numeric': ' - dd/mm/yyyy'
+                }
+            },
+            'en-us': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'MM d, Y',
+                    'numeric': 'mm/dd/yyyy - '
+                }
+            },
+            'exclaim': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': '\'!\'M Y',
+                    'long': '\'!\'MM d, Y',
+                    'numeric': '\'!\'mm/dd/yyyy - '
+                }
+            }
+        };
+
+        var module = {
+            getLocalizedDateString: getLocalizedDateString
+        };
+        return module;
+
+        function getLocalizedDateString(date, format, includeTime) {
```

Comment:
This could use a docstring since I expect it is going to be pretty heavily utilized as we add further localization.